lib/modulesFromList.nix

```diff
@@ -43,14 +35,40 @@ let
 
       genAttrs' = func: values: builtins.listToAttrs (map func values);
 
+      hasFileAttr = o: builtins.hasAttr "_file" o;
+      peak = f: f (builtins.functionArgs f);
+
     in
 
     genAttrs'
-      (path: {
-        name = removeSuffix ".toml" (removeSuffix ".nix" (baseNameOf path));
-        value = _import path;
-      })
-      paths;
+      (arg:
+
+        # a regular path to be imported
+        if builtins.isPath arg then
+          {
+            name = removeSuffix ".nix" (baseNameOf arg);
+            value = import arg;
+          }
+
+        # a module function with a _file attr
+        elseif ((builtins.isFunction arg) && (hasFileAttr (peek args)))
```

Comment:
Didnt know you could do else if like this, I thought they had to seperated in nix.